Sources/CALayer.swift

```diff
@@ -132,12 +132,6 @@ open class CALayer {
         willSet(newBounds) {
             guard newBounds != bounds else { return }
             onWillSet(keyPath: .bounds)
-
-            if bounds.size != newBounds.size {
-                // It seems weird to access the superview here but it matches the iOS behaviour
-                (self.superlayer?.delegate as? UIView)?.setNeedsLayout()
```

Comment:
`setsNeedLayout` should not be the cause of hundreds of re-renders – it deliberately only does layout once per frame.

the reason we do this is because that's what's happens on iOS – removing it would not be correct and would break other things.

I would suggest looking into calling `round` on the dimensions in the CGSize equality check – if we keep landing in this condition it's probably because of some floating point inaccuracies

Reply:
ok i think i see what you mean though – if we change the bounds twice in a render frame, even if the second `bounds` is equal to what it was at the start of the frame, we will still run layout

Reply:
so instead of comparing `newBounds` to the existing `bounds` (which could have just been set) we kind of need to save the bounds from the previous render cycle. doing this would be easy, but doing it well would take some thought – would you be interested in doing some brainstorming about this?

Reply:
sure! brainstorming sounds good @ephemer 

What you said is exactly whats's happening I think, and we really see that the relayouting happens for several frames because `setNeedsLayout` gets set to true over and over again and then it stops at some point (maybe when frames start to get skipped and then the layouting logic is able to catch up? I don't know yet, let's find it out?